Submitted By: Nick Fotopoulos <weasel at beyondnormal dot org>
Date: 2004-09-30
Initial Package Version: 0.3.3
Upstream Status: None
Origin:	Self
Description:	Updates libexif-gtk to use new exif_entry_get_value
		call from libexif-0.6.10.  The for takes three parameters
		instead of one like before.

diff -Naur libexif-gtk-0.3.3.orig/libexif-gtk/gtk-exif-content-list.c libexif-gtk-0.3.3/libexif-gtk/gtk-exif-content-list.c
--- libexif-gtk-0.3.3.orig/libexif-gtk/gtk-exif-content-list.c	2002-12-15 06:35:44.000000000 -0500
+++ libexif-gtk-0.3.3/libexif-gtk/gtk-exif-content-list.c	2004-09-30 18:21:07.571172520 -0400
@@ -368,12 +368,13 @@
 {
 	ExifEntry *e = data;
 	GValue value = {0};
+	char *evalue;
 
 	gtk_tree_model_get_value (model, iter, ENTRY_COLUMN, &value);
 	if (g_value_peek_pointer (&value) == e) {
 		g_value_unset (&value);
 		gtk_list_store_set (GTK_LIST_STORE (model), iter,
-				VALUE_COLUMN, exif_entry_get_value (e), -1);
+				VALUE_COLUMN, exif_entry_get_value (e, evalue, e->size), -1);
 		return (TRUE);
 	}
 	g_value_unset (&value);
@@ -394,13 +395,14 @@
 gtk_exif_content_list_add_entry (GtkExifContentList *list, ExifEntry *entry)
 {
 	GtkTreeIter iter;
+	char *entryvalue;
 
 	g_return_if_fail (GTK_EXIF_IS_CONTENT_LIST (list));
 
 	gtk_list_store_append (list->priv->store, &iter);
 	gtk_list_store_set (GTK_LIST_STORE (list->priv->store), &iter,
 			NAME_COLUMN, exif_tag_get_name (entry->tag),
-			VALUE_COLUMN, exif_entry_get_value (entry),
+			VALUE_COLUMN, exif_entry_get_value (entry, entryvalue, entry->size),
 			ENTRY_COLUMN, entry, -1);
 	g_signal_emit (list, signals[ENTRY_ADDED], 0, entry);
 }
diff -Naur libexif-gtk-0.3.3.orig/libexif-gtk/gtk-exif-entry-generic.c libexif-gtk-0.3.3/libexif-gtk/gtk-exif-entry-generic.c
--- libexif-gtk-0.3.3.orig/libexif-gtk/gtk-exif-entry-generic.c	2002-12-15 06:35:45.000000000 -0500
+++ libexif-gtk-0.3.3/libexif-gtk/gtk-exif-entry-generic.c	2004-09-30 18:23:17.433430456 -0400
@@ -89,6 +89,7 @@
 	GtkExifEntryGeneric *entry;
 	GtkWidget *table, *label;
 	gchar *txt;
+	char *evalue;
 
 	g_return_val_if_fail (e != NULL, NULL);
 
@@ -136,7 +137,7 @@
 	label = gtk_label_new ("Value:");
 	gtk_widget_show (label);
 	gtk_table_attach (GTK_TABLE (table), label, 0, 1, 3, 4, 0, 0, 0, 0);
-	label = gtk_label_new (exif_entry_get_value (e));
+	label = gtk_label_new (exif_entry_get_value (e, evalue, e->size));
 	gtk_widget_show (label);
 	gtk_table_attach (GTK_TABLE (table), label, 1, 2, 3, 4, 0, 0, 0, 0);
 
